aws/protocol/ec2query/error_utils_test.go

```diff
@@ -0,0 +1,41 @@
+package ec2query
+
+import (
+	"bytes"
+	"io"
+	"strings"
+	"testing"
+)
+
+func TestGetResponseErrorCode(t *testing.T) {
+	cases := map[string]struct {
+		errorResponse          io.Reader
+		expectedErrorCode      string
+	}{
+		"Invalid Greeting": {
+			errorResponse: bytes.NewReader([]byte(`<Response>
+			    <Errors>
+			        <Error>
+			            <Code>InvalidGreeting</Code>
+			            <Message>Hi</Message>
```

Comment:
Any particular reason to not try to extract the message?

Reply:
Here the intention was to only extract the error code to be able to direct the error response to correct error handling function specific to that error code.

But yeah I see the use-case where if the message is set and the error code is unknown ie. an error is unmodeled, we should be retrieving this message.

Is it safe to assume this Message is the errorMessage that we want to send for Unmodeled Errors?

Reply:
I think this might need change in XML too.

Reply:
Yeah, Message for sure. If you can make it case insensitive that's even better